### PR TITLE
ENH: FFT improvements. Closes #389

### LINKED ISSUE #389: Fourier transform from real to complex space has no adjoint
This error pops up any time you attempt to use the Fourier transform in an inverse problem, basically rendering it useless except in pre-post processing. This is a shame considering how cool it is :*(

The adjoint exists in _some_ sense mathematically. What steps do we need to take to make this workable?

### PR DESCRIPTION
**Pull request by adler-j**
* Allow adjoint on real transform
* Add test for the above
* Improve test structure in some fourier tests
* Change fftw specific functions to throw if impl is not fftw

### CODE REVIEW

File: odl/trafos/fourier.py [hunk 1]
```diff
@@ -2002,10 +2057,10 @@ def __init__(self, ran, dom=None, impl='numpy', **kwargs):
             of frequencies. It is guaranteed to be one-to-one
             (invertible).
 
-        * The `Operator.domain` of this operator always has the
+        * The `Operator.range` of this operator always has the
           `ComplexNumbers` as `LinearSpace.field`, i.e. if the
-          field of ``ran`` is the `RealNumbers`, this operator has no
-          `Operator.adjoint`.
+          field of ``dom`` is the `RealNumbers`, this operators adjoint is
```
Comment (kohr-h) [inline odl/trafos/fourier.py]: operator's

File: odl/trafos/fourier.py [hunk 2]
```diff
@@ -2002,10 +2057,10 @@ def __init__(self, ran, dom=None, impl='numpy', **kwargs):
...
+          defined by identifying real and complex parts.
```
Comment (kohr-h) [inline odl/trafos/fourier.py]: This makes no sense. You mean "defined by identifying real and imaginary parts with the components of a real product space element."
We should also have a section in the math documentation so we can refer to it here.

**Comment by kohr-h** [comment]
I'm good with all the changes, but as said, some words on the adjoint identification in the math part of the doc would be great.

File: odl/trafos/fourier.py [hunk 3]
```diff
@@ -835,9 +836,23 @@ def init_fftw_plan(self, planning_effort='measure', **kwargs):
...
+
+        Raises
+        ------
+        ValueError
+            If `impl` is not 'pyfftw'
+
+        Notes
+        -----
+        To save memory, clear the plan when the transform is no longer
+        used (the plan stores 2 arrays).
```
Comment (kohr-h) [inline odl/trafos/fourier.py]: Good to give the user a hint

**Comment by kohr-h** [comment]
You also need to update the Notes section of `FourierTransform` and the inverse (currently says "has no adjoint").

**Comment by adler-j** [comment]
Quite honestly I'm quite rusty on the maths here, would you like to write that short note @kohr-h ?

**Comment by kohr-h** [comment]
> Quite honestly I'm quite rusty on the maths here, would you like to write that short note @kohr-h ?

Ok alright.

**Comment by kohr-h** [comment]
> Quite honestly I'm quite rusty on the maths here, would you like to write that short note @kohr-h ?
> 
> Ok alright.

Done

**Comment by adler-j** [comment]
Merge ready?

**Pull request merged**